lib/elastic_record/as_document.rb

```diff
@@ -4,7 +4,7 @@ def as_search_document(mapping_properties = elastic_index.mapping[:properties])
       mapping_properties.each_with_object({}) do |(field, mapping), result|
         value = value_for_elastic_search field, mapping, mapping_properties
 
-        unless value.nil?
+        if !value.nil? || (respond_to?(:provided?) && provided?(field))
           result[field] = value
```

Comment:
Do you understand why we have separate as_search_document and as_partial_update_document functions?

Reply:
Presumably because you have a search/source/document representation and a need to build a submission/update. 

We can change the definition of `value_for_elastic_search_object` so it doesn't delegate to `as_search_document`, and put this behavior elsewhere.

But na, not concretely, that's the point of tagging people for review with many years of experience in a given repo.

Reply:
Partial updates is not just not-synonymous, it's a [distinct ElasticSearch feature](https://www.elastic.co/guide/en/elasticsearch/guide/current/update-doc.html) that we are opting to use.  

> We can change the definition of value_for_elastic_search_object so it doesn't delegate to as_search_document, and put this behavior elsewhere.

If you are proposing changing  as_partial_update_document to call itself recursively via value_for_elasticsearch_object, I think that is the right approach.

Reply:
I should clarify, I was aware of the Elastic Search api/feature. It's what we were debugging/working backgrounds from when we found this. I just meant that I wasn't necessarily aware of all the semantics within this repo/gem specifically.